### PR TITLE
[5.0] MailTemplate: Prevent notice in replaceTags

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
When registering a user, we export the whole data set of the user object to be used in the mail templates. Some of those values are actually NULL, for example `isRoot` or `lastResetTime`, etc. In line 350, we are doing a str_replace() on the tag with the value, in this case NULL. In PHP 8.1 (maybe even 8.0, didn't test.) this throws a notice, since str_replace() shouldn't be called with a NULL value.

I thought about how to solve the problem. The first idea was to just hopp over such an empty value, but that would leave a potential template tag in the mail. So I opted for replacing the value from NULL to an empty string.

### Testing Instructions
Can we do this with a code review? It is difficult to provide testing instructions without fabricating a component to display the issue and not have it hidden in a redirect...

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Mail/MailTemplate.php [hunk 1]
```diff
@@ -325,6 +325,11 @@ public function send()
     protected function replaceTags($text, $tags)
     {
         foreach ($tags as $key => $value) {
+            // If the value is NULL, we want to replace with an empty string. NULL itself throws notices
```
Comment (Quy) [inline libraries/src/Mail/MailTemplate.php]:
```suggestion
            // If the value is NULL, replace with an empty string. NULL itself throws notices
```

**Comment by HLeithner** [comment]
why is this a draft?

**Comment by Hackwar** [comment]
Because I didn't get around to write the test instructions yet.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3141e13b0677b4aea32ec031ddb958c6b7386a3d<br /><br />Before PR:
`PHP Deprecated:  str_replace(): Passing null to parameter #2 ($replace) of type array|string is deprecated in \libraries\src\Mail\MailTemplate.php on line 350`<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41679">issues.joomla.org/tracker/joomla-cms/41679</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 3141e13b0677b4aea32ec031ddb958c6b7386a3d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41679">issues.joomla.org/tracker/joomla-cms/41679</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41679">issues.joomla.org/tracker/joomla-cms/41679</a>.</sub>

**Comment by HLeithner** [comment]
thanks

**Pull request merged**